tools/s3.js

```diff
@@ -21,24 +21,22 @@ class s3 {
           reject(`error listing bucket objects ${err}`);
           return;
         }
-        
-        const items = data.Contents;
 
+        const items = data.Contents;
         if (items.length === 0) {
           resolve();
         }
 
+        const deleteItemsPromises = [];
         items.forEach((item, index) => {
           const deleteParams = { Bucket: this.bucket, Key: item.Key };
-          this.client.deleteObject(deleteParams, (err, data) => {
-            if (err) {
-              reject(`delete err ${deleteParams.Key}`);
-            }
-            if (index === items.length - 1) {
-              resolve();
-            }
+          const promise = new Promise((resolve,reject) => {
```

Comment:
remove `reject` param, you're not using it.

```suggestion
          const promise = new Promise((resolve) => {
```